sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -9,7 +9,60 @@ dotenv.config({ path: "../.env" });
 
 let nock: any;
 if (!isBrowser()) {
-  nock = require("nock");
+  if (process.env.TEST_MODE === "record" || process.env.TEST_MODE === "playback") {
+    nock = require("nock");
+  }
+} else if ((window as any).__env__.TEST_MODE === "record") {
+  // Converting content corresponding to all the console statements
+  //  into (JSON.stringify)-ed content in record mode for browser tests.
+  //
+  // In browser, once the content to be recorded is ready, recordings
+  //  are supposed to be sent to the appropriate karma reporter(jsonToFileReporter)
+  //  in order to generate the corresponding recording file.
+  // The way to do this is by printing the recordings as JSON strings to `console.log()`.
+  // As a result, the console gets filled with lots of prints while recording.
+  //
+  // We solve this issue by
+  // - disabling the console.logs from karma and
+  // - by adding a custom console.log() which converts all the console statements into
+  //   console.log() with stringified JSON objects.
+  // - Handle all the console.logs with stringified JSON objects in karma.conf.js
+  //   as explained below.
+  //
+  // Karma.conf.js
+  // - jsonToFileReporter in karma.conf.js filters the JSON strings in console.logs.
+  // - Console logs with `.writeFile` property are captured and are written to a file(recordings).
+  // - The other console statements are captured and printed normally.
+  // - Example - console.warn("hello"); -> console.log({ warn: "hello" });
+  // - Example - console.log("hello"); -> console.log({ log: "hello" });
+  const consoleLog = window.console.log;
+  for (const method in window.console) {
+    if (typeof (window.console as any)[method] === "function") {
```

Comment:
Shouldn't we make sure that `window.console` has its own `method` property (`hasOwnProperty`)? By skipping it we are [checking all of its prototype chain](https://stackoverflow.com/questions/9396569/javascript-what-is-property-in-hasownproperty).

Reply:
Added the check